staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go

```diff
@@ -36,6 +36,27 @@ func MakeUsername(namespace, name string) string {
 	return ServiceAccountUsernamePrefix + namespace + ServiceAccountUsernameSeparator + name
 }
 
+// MatchesUsername checks whether the provided username matches the namespace and name without
+// allocating. Use this when checking a service account namespace and name against a known string.
+func MatchesUsername(namespace, name string, username string) bool {
+	if !strings.HasPrefix(username, ServiceAccountUsernamePrefix) {
+		return false
+	}
+	username = strings.TrimPrefix(username, ServiceAccountUsernamePrefix)
+
+	if !strings.HasSuffix(username, name) {
```

Comment:
alternative prefix/suffix is hard to follow... consider checking prefix and consuming ServiceAccountUsernamePrefix, then namespace, then ServiceAccountUsernameSeparator, then name?

Reply:
Algorithmic improvement to check prefix then suffix.

Reply:
It might be a wash across namespaces, but in the common ones (all in the same namespace) it's a lot.  I won't go to the wall for it.

Reply:
if it makes a difference, I can live with it

Reply:
Actually, I did the simpler thing which bypasses TrimPrefix and is probably faster than both because we're doing a linear scan.